src/systems/interactions.js

```diff
@@ -5,28 +5,16 @@ import { waitForDOMContentLoaded } from "../utils/async-utils";
 import { canMove } from "../utils/permissions-utils";
 
 function findHandCollisionTargetForHand(body) {
-  const driver = AFRAME.scenes[0].systems.physics.driver;
-  const numManifolds = driver.dispatcher.getNumManifolds();
-  const handPtr = Ammo.getPointer(body);
-  for (let i = 0; i < numManifolds; i++) {
-    const persistentManifold = driver.dispatcher.getManifoldByIndexInternal(i);
-    const body0ptr = Ammo.getPointer(persistentManifold.getBody0());
-    const body1ptr = Ammo.getPointer(persistentManifold.getBody1());
-    if (handPtr !== body0ptr && handPtr !== body1ptr) {
-      continue;
-    }
-    const numContacts = persistentManifold.getNumContacts();
-    for (let j = 0; j < numContacts; j++) {
-      const manifoldPoint = persistentManifold.getContactPoint(j);
-      if (manifoldPoint.getDistance() <= 10e-6) {
-        const object3D = driver.els.get(handPtr === body0ptr ? body1ptr : body0ptr).object3D;
-        if (object3D.el && object3D.el.components.tags && object3D.el.components.tags.data.isHandCollisionTarget) {
-          return object3D.el;
-        }
-        return null;
-      }
+  const world = this.el.sceneEl.systems["hubs-systems"].physicsSystem.world;
+  const handPtr = Ammo.getPointer(body.physicsBody);
+
+  if (world.collisions.has(handPtr) && world.collisions.get(handPtr).length > 0) {
+    const object3D = world.object3Ds.get(world.collisions.get(handPtr)[0]);
```

Comment:
Since this is reasonably hot code, maybe it would be nicer to do less work here? e.g. 
```javascript
const handCollisions = world.collisions.get(handPtr);
if (handCollisions != null && handCollisions.length > 0) {
  const object3D = world.object3Ds.get(handCollisions[0]);
  // ...
}

Reply:
Is this code correct? What if the hand is colliding with more than one thing, but the first one you look at isn't a collision target? The old version would continue looking at the other things, but the new version gives up.